internal/controllers/aggregation/status.go

```diff
@@ -57,29 +59,52 @@ func (s *statusController) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		}
 
 		for _, state := range slice.Status.Resources {
+			state := state
 			// Sync
 			if !state.Reconciled || (comp.DeletionTimestamp != nil && !state.Deleted) {
 				reconciled = false
 			}
 
 			// Readiness
-			if state.Ready == nil || !*state.Ready {
+			if state.Ready == nil {
 				ready = false
 			}
+			if state.Ready != nil && (maxReadyTime == nil || maxReadyTime.Before(state.Ready)) {
+				maxReadyTime = state.Ready
+			}
 		}
 	}
 
-	if comp.Status.CurrentState.Reconciled == reconciled && comp.Status.CurrentState.Ready == ready {
+	if (comp.Status.CurrentSynthesis.Reconciled != nil) == reconciled && (comp.Status.CurrentSynthesis.Ready != nil) == ready {
```

Comment:
The conditionals are getting trickier and trickier to read 😄 we should probably to a cleanup pass later on

Reply:
Yeah fully agree. I'd like to refactor a bunch of the controller conditionals into pure funcs and write unit tests against them to capture the intention